portal_pages/tests/test_user_forms.py

```diff
@@ -0,0 +1,155 @@
+        self.marketplace_entry_form_data['company_company_email'] = 'Iamabot@bots.com'
+        resp = self.client.post(self.marketplace_submission_url, self.marketplace_entry_form_data)
+        self.assertEqual(resp.status_code, 200)
+        self.assertTrue('form' in resp.context)
+        form = resp.context['form']
+        self.assertEqual(len(form.errors), 1)
+        self.assertTrue('__all__' in form.errors)
+        self.assertEqual(form.errors['__all__'], ['Invalid submission'])
+        self.assertEqual(MarketplaceEntryPage.objects.all().count(), 0)
+
+    def test_case_study_form_valid(self):
+        # This should successfully add a single new case study
+        resp = self.client.post(self.case_study_submission_url, self.case_study_form_data)
+        self.assertEqual(resp.status_code, 302)
+        expected_redirect = self.case_study_index_page.url + self.case_study_index_page.reverse_subpage('thanks')
+        self.assertRedirects(resp, expected_redirect)
+        self.assertEqual(CaseStudyPage.objects.all().count(), 1)
+        case_study_page = CaseStudyPage.objects.all()[0]
+        # The original date should have been overriden by month_start/year_start
+        self.assertEqual(case_study_page.date, datetime.date(self.case_study_form_data['year_start'], self.case_study_form_data['month_start'], 1))
+
+    def test_case_study_form_invalid_honeypot(self):        
+        self.case_study_form_data['email_field_email'] = 'notnine'
+        resp = self.client.post(self.case_study_submission_url, self.case_study_form_data)
+        self.assertEqual(resp.status_code, 200)
+        self.assertTrue('form' in resp.context)
+        form = resp.context['form']
+        self.assertEqual(len(form.errors), 1)
+        self.assertTrue('__all__' in form.errors)
+        self.assertEqual(form.errors['__all__'], ['Invalid submission'])
+        self.assertEqual(MarketplaceEntryPage.objects.all().count(), 0)
+        self.assertEqual(CaseStudyPage.objects.all().count(), 0)
+
+    def test_case_study_invalid_bad_email(self):
+        self.case_study_form_data['submitter_email'] = 'invalid@nope'
+        resp = self.client.post(self.case_study_submission_url, self.case_study_form_data)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.context[0]['form'].errors['submitter_email'], ['Enter a valid email address.'])
+        self.assertEqual(CaseStudyPage.objects.all().count(), 0)
+
+    def test_blog_form_valid(self):
+        # This should successfully add a single new blog post
+        resp = self.client.post(self.blog_submission_url, self.blog_form_data)
+        self.assertEqual(resp.status_code, 302)
+        expected_redirect = self.blog_index_page.url + self.blog_index_page.reverse_subpage('thanks')
+        self.assertRedirects(resp, expected_redirect)
+        self.assertEqual(resp['Location'], 'http://testserver/%s/submit-thank-you/'% self.blog_index_page.slug)
```

Comment:
This is duplicating the assertRedirects test? I think this line should have been removed?